DSharpPlus/Net/Gateway/Compression/Zstd/ZstdDecompressor.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+
+using CommunityToolkit.HighPerformance;
+using CommunityToolkit.HighPerformance.Buffers;
+
+namespace DSharpPlus.Net.Gateway.Compression.Zstd;
+
+/// <summary>
+/// A payload decompressor using zstd.
+/// </summary>
+public sealed class ZstdDecompressor : IPayloadDecompressor
+{
+    private ZstdInterop wrapper = new();
```

Comment:
This class should probably have a finalizer that ensures the wrapper is released even when Dispose isn't called, ideally with enough redundancy to avoid double frees. It's likely not a big deal for this application, but may as well.